indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -149,13 +184,19 @@ export default function FormItem({
       </>
     ) : null;
 
+  if (!show && !setupMode) {
+    form.change(rest.htmlName, null);
+    return null;
+  }
+
   return (
     <div
       styleName={`form-item ${toClasses({
         'disabled': !isEnabled || paidItemLocked,
```

Comment:
Should it be possible to depend on a disabled field? Currently, it is allowed

Reply:
The default behaviour is to show the field if it depends on a disabled conditional field, since such field is not in the rendered form. I think that's the sensible default, but another option would be to break that "showIf" relationship whenever the field gets disabled. But that's a destructive operation, as if the field is enabled again the link will have already been destroyed.

Up to you guys. I honestly think this could be subject to experimentation and user's feedback to find out how they plan on using the feature, but if you have a clear vision on this matter I happily change it.

Reply:
FWIW in my case I ended up selecting the "Title" field (disabled by default) as a condition for the "Affiliation" field when starting to test this PR. I only realized afterwards, that "Title" is disabled.

I think in reality there won't be many disabled fields, but "Title" is the one that will almost always be there (except in events that actively decide to enable that field) - so IMHO it makes sense to not allow choosing this as a condition.

Reply:
Maybe we should simply now allow disabling/deleting a field that's a condition for another field? That way we can nicely fail, tell the user about it, and don't have to add any new UI since in the end it'll be just returning an error from the server.

Reply:
I was a bit tricky to detect whether a given field is being used as a conditional one, but I managed.

See [85e879a](https://github.com/indico/indico/pull/6678/commits/85e879a78cdd306f69141e622f945595217f461f).